operator/v1/types_dns.go

```diff
@@ -73,6 +83,29 @@ type ForwardPlugin struct {
 	Upstreams []string `json:"upstreams"`
 }
 
+// DNSNodePlacement describes the node scheduling configuration for DNS pods.
+type DNSNodePlacement struct {
+	// nodeSelector is the node selector applied to DNS pods.
+	//
+	// If unset, the default is the following:
+	//
+	//   beta.kubernetes.io/os: linux
+	//
+	// If set, the specified selector is used and replaces the default.
+	//
+	// +optional
+	NodeSelector *metav1.LabelSelector `json:"nodeSelector,omitempty"`
```

Comment:
pretty sure that pods are still restricted by `NodeSelector map[string]string` .  I suggest having this match that schema even though it is an older style selector.

Reply:
We use `metav1.LabelSelector` for the IngressController API, so that puts us in a slightly icky place:
https://github.com/openshift/api/blob/517b8a7697503a7502e50b2f87d6857099880504/operator/v1/types_ingress.go#L195-L206
But I guess it's better to do the right thing here than to be consistent with a bad API, right?

Reply:
> But I guess it's better to do the right thing here than to be consistent with a bad API, right?

I think so.  I would like to avoid the ability to have incompatible data if possible.  I should have caught it in that operator too.